seerep_srv/seerep_core_fb/src/core_fb_image.cpp

```diff
@@ -66,47 +66,49 @@ boost::uuids::uuid CoreFbImage::addData(const seerep::fb::Image& img)
   }
 }
 
-void CoreFbImage::addBoundingBoxesLabeled(const seerep::fb::BoundingBoxes2DLabeledStamped& boundingBoxes2dlabeled)
+void CoreFbImage::addLabel(const seerep::fb::DatasetUuidLabel& datasetUuidLabel)
 {
   boost::uuids::string_generator gen;
-  boost::uuids::uuid uuidMsg = gen(boundingBoxes2dlabeled.header()->uuid_msgs()->str());
-  boost::uuids::uuid uuidProject = gen(boundingBoxes2dlabeled.header()->uuid_project()->str());
+  boost::uuids::uuid uuidMsg = gen(datasetUuidLabel.datasetUuid()->str());
+  boost::uuids::uuid uuidProject = gen(datasetUuidLabel.projectUuid()->str());
 
   auto hdf5io = CoreFbGeneral::getHdf5(uuidProject, m_seerepCore, m_hdf5IoMap);
 
-  hdf5io->writeImageBoundingBox2DLabeled(boost::lexical_cast<std::string>(uuidMsg), boundingBoxes2dlabeled.labels_bb());
+  hdf5io->writeLabelsFb(boost::lexical_cast<std::string>(uuidMsg), datasetUuidLabel.datasetUuid()->str(),
+                        datasetUuidLabel.labels());
 
```

Comment:
For one UUID a cast is used, the other is simply extracted from the method parameter.